redaxo/src/core/lib/view.php

```diff
@@ -320,7 +320,10 @@ public static function title($head, $subtitle = null)
         $fragment->setVar('subtitle', $subtitle, false);
         $return = $fragment->parse('core/page/header.php');
 
-        $return .= rex_extension::registerPoint(new rex_extension_point('PAGE_TITLE_SHOWN', ''));
+        /** @psalm-taint-escape html */ // we expect safe content from the extensions
+        $afterTitle = rex_extension::registerPoint(new rex_extension_point('PAGE_TITLE_SHOWN', ''));
```

Comment:
Ich kapiere noch nicht, wie ein so eine Stelle so einen Unterschied machen kann.
Aber diese Änderung bringt mich lokal runter von 328 auf 270 Fehler.

Ich denke, diese Annahme hier so zu treffen, ist ok, oder?
Also dass wir sagen, Extension zu `PAGE_TITLE_SHOWN` müssen selbst dafür sorgen, sicheren Content zu liefern.

Reply:
> Ich kapiere noch nicht, wie ein so eine Stelle so einen Unterschied machen kann.
> Aber diese Änderung bringt mich lokal runter von 328 auf 270 Fehler.

ich denke es ist nicht die Änderung des php-codes der die warnungen runter bringt, sonder das "stumm"-schalten von psalm durch den kommentar.

vermutlich nutzen wir diese funktion in fast allen pages-dateien und daher liefert es dann 1 warnung je aufruf von `rex_view::title()`.



> Ich denke, diese Annahme hier so zu treffen, ist ok, oder?
> Also dass wir sagen, Extensions zu `PAGE_TITLE_SHOWN` müssen selbst dafür sorgen, sicheren Content zu liefern.

es gibt nur 2 Möglichkeiten:
1) man erlaubt kein html und escaped hier schon im core.
2) man erlaubt dem EP html zu rendern. wenn er das tut muss er auch immer escapen. wer rendert muss immer auch escapen.

Reply:
ggf macht es sinn mal durchzuschauen wie der EP dann genutzt wird: https://github.com/search?q=org%3AFriendsOfREDAXO+PAGE_TITLE_SHOWN&type=code

Reply:
nochmal drüber nachgedacht: ich denke hier ist ein psalm bug. wenn psalm denkt dass wir hier user-provided input verarbeiten würden, müsste es ja irgendwo in der codebase eine extension geben die user input hier reinreicht. 

an dieser stelle hier jetzt `@psalm-taint-escape html` zu notieren, versteckt nur die meldung aber löst das problem nicht, falls es wirklich so ist.